transport/spi/src/main/kotlin/EndpointComponent.kt

```diff
@@ -68,9 +68,22 @@ abstract class EndpointComponent<T : Any>(
         private val logger: Logger = LoggerFactory.getLogger(EndpointComponent::class.java)
 
         /**
-         * Get the keep-alive file in the user's home directory.
+         * Get the keep-alive file in the user's home directory. As a fallback, if the user's home directory ist not
+         * writable, use the system's temporary directory.
          */
-        private fun getKeepAliveFile(): File = File(System.getProperty("user.home"), KEEP_ALIVE_FILE_NAME)
+        private fun getKeepAliveFile(): File {
+            val keepAliveFile = File(System.getProperty("user.home"), KEEP_ALIVE_FILE_NAME)
+            return if (keepAliveFile.parentFile.canWrite()) {
```

Comment:
How about extracting `val userHome = File(System.getProperty("user.home"))` so you don't have to "re-determine" it via `keepAliveFile.parentFile`?

Reply:
In total, this could look like

```kotlin
private fun getKeepAliveFile(): File {
    val userHome = File(System.getProperty("user.home"))
    return if (userHome.canWrite()) {
        userHome.resolve(KEEP_ALIVE_FILE_NAME)
    } else {
        val tmpDir = System.getProperty("java.io.tmpdir")
        tmpDir.resolve(KEEP_ALIVE_FILE_NAME).also {
            logger.info(
                "Home directory is not writable. Using temporary directory for keep-alive file: " +
                    "${it.absolutePath}"
            )
        }
    }
}
```